encoding/ipfix/decode.go

```diff
@@ -3,90 +3,127 @@ package ipfix
 import (
 	"bytes"
 	"encoding/binary"
-	"errors"
 	"io"
+	"reflect"
 )
 
 // TODO: what about padding (less bytes in data structures than declared message size)?
 
+// An InvalidUnmarshalError describes an invalid argument passed to Unmarshal.
+// (The argument to Unmarshal must be a non-nil pointer.)
+type InvalidUnmarshalError struct {
+	Type reflect.Type
+}
+
+func (e *InvalidUnmarshalError) Error() string {
+	if e.Type == nil {
+		return "ipfix: Unmarshal(nil)"
+	}
+
+	if e.Type.Kind() != reflect.Ptr {
+		return "ipfix: Unmarshal(non-pointer " + e.Type.String() + ")"
+	}
+
+	return "ipfix: Unmarshal(nil " + e.Type.String() + ")"
+}
+
+type Decode struct {
```

Comment:
Codacy found an issue: [exported type Decode should have comment or be unexported](https://app.codacy.com/manual/mateumann/clouseau/pullRequest?prid=6326016)